### PR TITLE
Updates developer ci & Adds MNIST build test

### PR DESCRIPTION
**Pull request by srcansiz**
**MR description**

- Remove Jenkins form CI docs and adds  Github actions 
- Adds run_test_mnist to Fed-BioMed build test 

 
**Developer Certificate Of Origin (DCO)**

By opening this merge request, you agree to the
[Developer Certificate of Origin (DCO)](https://github.com/fedbiomed/test-fedbiomed/blob/develop/CONTRIBUTING.md#fed-biomed-developer-certificate-of-origin-dco)

This DCO essentially means that:

- you offer the changes under the same license agreement as the project, and
- you have the right to do that,
- you did not steal somebody else’s work.

**License**

Project code files should begin with these comment lines to help trace their origin:
```
# This file is originally part of Fed-BioMed
# SPDX-License-Identifier: Apache-2.0
```

Code files can be reused from another project with a compatible non-contaminating license.
They shall retain the original license and copyright mentions.
The `CREDIT.md` file and `credit/` directory shall be completed and updated accordingly.

**Guidelines for MR review**

General:

* give a glance to [DoD](https://fedbiomed.org/latest/developer/Fed-BioMed_DoD.pdf)
* check [coding rules and coding style](https://fedbiomed.org/latest/developer/usage_and_tools/#coding-style)
* check docstrings (eg run `tests/docstrings/check_docstrings`)

Specific to some cases:

* update all conda envs consistently (`development` and `vpn`, Linux and MacOS)
* if modified researcher (eg new attributes in classes) check if breakpoint needs update (`breakpoint`/`load_breakpoint` in `Experiment()`, `save_state`/`load_state` in aggregators, strategies, secagg, etc.)
* if modified a component with versioning (config files, breakpoint, messaging protocol) then update the version following the rules in `common/utils/_versions.py`

### CODE REVIEW

File: .github/workflows/build-test.yml [hunk 1]
```diff
@@ -105,4 +102,40 @@ jobs:
...
+              - 'envs/development/docker/**'
+              - 'notebooks/**'
+              - '.github/**'
+
+      - name: Run MNIST Test - Master
+        if: steps.filter.outputs.module == 'true'
+        run: |
+          source ~/.bashrc
+          ./scripts/fedbiomed_run network
+          ./scripts/run_test_mnist data
```
Comment (mvesin) [inline .github/workflows/build-test.yml:135]: If we call call `run_test_mnist`, then `CI-build` can be removed => doing it

**Comment by mvesin** [comment]
hi @srcansiz 

In `.github/actions/unit-tests/actions.yml` I suspect we should also configure conda & start the `network` component. If not, it won't be updated as needed + restarted after a test that stops it.

```
runs:
  using: "composite"
  steps:
    - name: Install researcher dependencies
      run: | 
        source ~/.bashrc
        ./scripts/configure_conda researcher network
        ./scripts/fedbiomed_run network
      shell: bash
```

This is how it behaves under gitlab.

File: .github/workflows/build-test.yml [hunk 2]
```diff
@@ -105,4 +102,40 @@ jobs:
...
+              - 'envs/development/conda/**'
+              - 'envs/development/docker/**'
+              - 'notebooks/**'
+              - '.github/**'
+
+      - name: Run MNIST Test - Master
+        if: steps.filter.outputs.module == 'true'
+        run: |
+          source ~/.bashrc
+          ./scripts/fedbiomed_run network
```
Comment (mvesin) [inline .github/workflows/build-test.yml:134]: As we suppressed the use of `./scripts/CI_build` that was doing a `./scripts/configure_conda` I suspect we need at least a : `./scripts/configure_conda node` (network and researcher were done by unit tests ...).

If not, the conda envs are not properly updated.

Comment (srcansiz) [reply .github/workflows/build-test.yml:134]: Hi @mvesin, 
Thanks for reminding. I am going to add it.

**Comment by srcansiz** [comment]
> hi @srcansiz
> 
> In `.github/actions/unit-tests/actions.yml` I suspect we should also configure conda & start the `network` component. If not, it won't be updated as needed + restarted after a test that stops it.
> 
> ```
> runs:
>   using: "composite"
>   steps:
>     - name: Install researcher dependencies
>       run: | 
>         source ~/.bashrc
>         ./scripts/configure_conda researcher network
>         ./scripts/fedbiomed_run network
>       shell: bash
> ```
> 
> This is how it behaves under gitlab.

Hi @mvesin, 

Is network component is used in unit tests? I don't think starting network component should be part of unit test.

**Comment by mvesin** [comment]
Hi @srcansiz 

> Is network component is used in unit tests? I don't think starting network component should 
> be part of unit test.

Yes, network component is needed, see #390 
So we need `configure_conda network` and `fedbiomed_run network` => restoring this

**Review by mvesin — APPROVED** [review]

**Pull request merged**